### PR TITLE
Use master environment variable from KubernetesClusterSchedulerBackend

### LINKED ISSUE #112: java.net.UnknownHostException: kubernetes in KubernetesClusterSchedulerBackend for Minikube
While working on getting Ingresses up in #70, I had to upgrade Minikube. I upgraded it to 0.16.0, and started seeing this error in KubernetesClusterSchedulerBackend:

`java.net.UnknownHostException: kubernetes: Try again`

Looks like setting the API server url to `https://kubernetes` doesn't work in this environment. We do this here: https://github.com/apache-spark-on-k8s/spark/blob/k8s-support-alternate-incremental/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala#L47

### PR DESCRIPTION
**Pull request by mccheah**
Closes #112.

### CODE REVIEW

**Comment by mccheah** [comment]
@foxish this fixes the issue on my machine.

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesClientBuilder.scala [hunk 1]
```diff
@@ -32,12 +34,13 @@ private[spark] object KubernetesClientBuilder {
...
-      kubernetesMaster: String,
-      kubernetesNamespace: String): DefaultKubernetesClient = {
+  def buildFromWithinPod(kubernetesNamespace: String, useSsl: Boolean): DefaultKubernetesClient = {
+    val kubernetesHost = System.getenv(ENV_KUBERNETES_SERVICE_HOST)
+    val kubernetesPort = System.getenv(ENV_KUBERNETES_SERVICE_PORT)
+    val urlScheme = if (useSsl) "https" else "http"
     var clientConfigBuilder = new ConfigBuilder()
       .withApiVersion("v1")
-      .withMasterUrl(kubernetesMaster)
+      .withMasterUrl(s"$urlScheme://$kubernetesHost:$kubernetesPort")
```
Comment (foxish) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesClientBuilder.scala:43]: This should be `https` always. Even if the user uses the insecure endpoint to access the apiserver from outside the cluster, KUBERNETES_SERVICE_PORT should point to the secure endpoint.

**Comment by foxish** [comment]
I'm a bit wary about this change because the most robust way here is to use the dns name and let it be resolved. 
There are other issues around using the pod env vars, like https://github.com/kubernetes/kubernetes/issues/40973

The comments in https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L412-L416 read:
```
	// Note that there is a race between Kubelet seeing the pod and kubelet seeing the service.
	// To avoid this users can: (1) wait between starting a service and starting; or (2) detect
	// missing service env var and exit and be restarted; or (3) use DNS instead of env vars
	// and keep trying to resolve the DNS name of the service (recommended).
```

Have you tried resolving the DNS name in a loop within the driver pod? Does it always fail in your environment?

**Comment by mccheah** [comment]
The issue has reproduced every time I've run the integration tests. I haven't tried resolving in a loop yet.

**Comment by foxish** [comment]
If it is specific to minikube 0.16.0, I think we should report it and switch to the newer version when it's fixed.

**Comment by mccheah** [comment]
I also saw it in Minikube 0.15.0. I also just tried resolving in a loop - basically trying the failing call to `kubernetesClient.pods()....` 5 times with 5 second intervals in `KubernetesClusterSchedulerBackend`. The resolution failed every time.

**Comment by foxish** [comment]
I wasn't testing cross namespace. The issue was that it is no longer automatically looking up `kubernetes.svc.default` in a different namespace. I added a fix in https://github.com/apache-spark-on-k8s/spark/pull/118. PTAL. 

I should've caught this earlier. Apologies.

**Comment by mccheah** [comment]
Closing in favor of #118 which I've confirmed fixes the issue.

**Pull request closed without merging**